src/main/resources/lib/enonic/static/resource/path/mappedRelativePath.ts

```diff
@@ -0,0 +1,8 @@
+import type {Request} from '/lib/enonic/static/types';
+
+import {getRelative} from '/lib/enonic/static/resource/path/getRelative';
+
+export const mappedRelativePath =
+  (base: string) =>
+  (request: Request): string =>
+    getRelative(request).substring(base.length);
```

Comment:
pretty simple implementation, but does not assert that path is correct. I think it should error out if substring does not match base - to avoid security issues

Reply:
Yeah, it was beautiful, but unsafe. New one pushed.